src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -53,7 +56,20 @@ std::string GetFullType(const char * type, DnssdServiceProtocol protocol)
     std::ostringstream typeBuilder;
     typeBuilder << type;
     typeBuilder << (protocol == DnssdServiceProtocol::kDnssdProtocolUdp ? kProtocolUdp : kProtocolTcp);
-    return typeBuilder.str();
+
+    auto fullType                = typeBuilder.str();
+    std::string subtypeDelimiter = "._sub.";
+    size_t position              = fullType.find(subtypeDelimiter);
+    if (position != std::string::npos)
```

Comment:
So did `OnBrowseRemove` use to do the wrong thing in its type compare?  And did `Resolve` just use to work because there were never subtypes involved there?  It seems like this is effectively changing both of those callsites....

Reply:
I agree that the original intention is unclear here. Initially the issue is that the "type" passed in as a parameter to `ChipDnssdBrowse" is formatted in a way that does not fit the expected format under darwin. 
It should not affects results retrieved into `OnBrowseRemove`.
And for the `Resolve` it does not matter because we used to not use subtypes.

To make that clearer I have added an other method called "GetFullTypeWithSubTypes" (same name as the one used for "Register") and I have used it in the "Browse" case, while "Resolve" uses "GetFullType".